### PR TITLE
Auto align

### PR DESCRIPTION
**Pull request by ky28059**
Also includes a reabstraction of `PlacePosition` to use two `FieldPosition`s instead of reimplementing `FieldPosition`'s pose logic and abstracting the initial piece placement of the preloaded piece into `BaseAutonSequence`'s constructor.

### CODE REVIEW

File: src/main/java/frc/robot/commands/auton/BaseAutonSequence.java
Comment (e3l) [inline src/main/java/frc/robot/commands/auton/BaseAutonSequence.java:86]: why not just use a PlacePosition?

**Review by e3l — COMMENTED** [review]
lgtm

File: src/main/java/frc/robot/commands/swerve/GoToPointCommand.java
Comment (e3l) [inline src/main/java/frc/robot/commands/swerve/GoToPointCommand.java:50]: it might be worth adding a (configurable in the constructor) time kill switch also. if the driver can seamlessly take over it might not be useful in auto-aligning, but if we use this in auto, we'll probably want one

File: src/main/java/frc/robot/commands/dropping/AutoAlignCommand.java
Comment (e3l) [inline src/main/java/frc/robot/commands/dropping/AutoAlignCommand.java:178]: can we keep the current target elevator state instead? if the elevator is travelling to another state, it doesn't make sense to me to just take the closest one instead of getting to existing target

**Comment by e3l** [comment]
to test thoroughly before merging this PR:
- auto aligning
- all utah autons: balancing (no mobility), one piece + taxi

File: src/main/java/frc/robot/commands/dropping/AutoAlignCommand.java
Comment (e3l) [inline src/main/java/frc/robot/commands/dropping/AutoAlignCommand.java:100]: "left" and "right" are probably misleading now. flipping based on alliance color doesn't seem to be implemented yet, either

File: src/main/java/frc/robot/commands/dropping/AutoAlignCommand.java
Comment (e3l) [inline src/main/java/frc/robot/commands/dropping/AutoAlignCommand.java:150]: this doesn't seem to be the most clean solution, but the alternative might be a big change. i'm thinking a class containing two enums. this would allow us to make methods that return a new instance of the class with one of the two enums different -- essentially, decouple the two states we're trying to decouple

**Review by e3l — APPROVED** [review]
lgtm, if not urgent:
- have we tested cancelling?
- need to test autons

known issues:
- photonvision config not persist

File: src/main/java/frc/robot/commands/swerve/GoToPointCommand.java
Comment (ky28059) [reply src/main/java/frc/robot/commands/swerve/GoToPointCommand.java:72]: Hmm, my thinking was that that would require two more overloads to be written for `.setDrivePowers` and `.setDrivePowersWithHeadingLock` and maybe it would make more sense for those two methods to act as the driver interface only (`FollowPathCommand` already uses `.setModuleStates()`, and the fact that `.setDrivePowers` takes [-1.0, 1.0] powers instead of velocities makes them more suited for joystick input than PID output)

**Comment by ky28059** [comment]
Cancelling with the button definitely works, but unsure if Ryan used the joystick to cancel today.

File: src/main/java/frc/robot/commands/auton/BalanceAndTaxiAutonSequence.java
Comment (CrolineCrois) [inline src/main/java/frc/robot/commands/auton/BalanceAndTaxiAutonSequence.java:13]: taxi and balance

**Comment by e3l** [comment]
tested balance & bottom blue autons

File: src/main/java/frc/robot/commands/auton/BalanceAndTaxiAutonSequence.java
Comment (e3l) [reply src/main/java/frc/robot/commands/auton/BalanceAndTaxiAutonSequence.java:13]: yes true can u do this on ur branch

**Pull request merged**
